### PR TITLE
Use less derived Collider2D where possible

### PR DESCRIPTION
**Pull request by flibber-hk**
This has no change in isolation - for a BoxCollider2D x.bounds.size and x.size return the same Vector2.

### CODE REVIEW

**Comment by Extremelyd1** [comment]
Could you perhaps explain why this change is required/desirable?

**Comment by flibber-hk** [comment]
My SkillUpgrades mod needs to replace the knight's collider with a [polygon collider](https://github.com/flibber-hk/HollowKnight.SkillUpgrades/blob/main/SkillUpgrades/Util/HeroRotation.cs#L21) to function properly - that was the easiest way I found to rotate the knight while keeping the hitbox in place. Unfortunately, this means that HKMP fails to create player prefabs because `localPlayerObject.GetComponent<BoxCollider2D>()` returns null.

I think it would be correct that the fix *should* come on my end; however, that fix would be writing an IL hook to patch your method, which would be fairly bothersome compared to the two line C\# fix. (Actually the ideal fix would be an API allowing me to modify the SpawnPlayer method, but to be honest I don't think there's a sensible way to do this without forcing me to write an IL or On hook, which I may end up having to do anyway when the main HKMP API is published to give the other players a polygon collider hitbox.)

TL;DR it's my responsibility to fix rather than yours, but I'd prefer not to write a complicated 30 line fix on my end when a trivial 2 line fix exists on yours 🤷 if you'd prefer me to do it on my end then I probably could though

Side note - it's not actually dishonest that the player has a PolygonCollider locally while a BoxCollider remotely, because when rotating the player's hitbox stays in place locally and the player simply doesn't rotate at all remotely.

**Comment by flibber-hk** [comment]
(I'll also note that it's not really problematic from a vanilla game perspective that I'm replacing the collider with a polygon collider, because all internal references to the knight's hitbox use Collider2D rather than BoxCollider2D.)

**Comment by Extremelyd1** [comment]
The changes you propose are fine and, as you mention, only have impact outside this codebase. I was mainly asking for a motivation, because I couldn't deduce it from the code.
